tests/php/modules/shortcodes/test-class.slideshow.php

```diff
@@ -48,34 +57,72 @@ public function test_shortcodes_slideshow() {
+			),
```

Comment:
Those 2 tests fail on WordPress.com, because the images aren't the ones you're expecting:
https://github.com/Automattic/jetpack/blob/a952f7254a18c8e10ad504102c168d5a9dfaa97c/tests/php/modules/shortcodes/test-class.slideshow.php#L6-L10

I think you may need to update those tests to be a less exact maybe, to take into account different images?

Reply:
Hi Jeremy!
Ah, good point. Sorry for the delay.

The [commits below](https://github.com/Automattic/jetpack/compare/230427c8af274e8f2de95dbedf5ae1126d71ee01...6bc182685ea157c20b6095932f9684669e07bef4) interpolate the image URLs instead of hard-coding them. Also, they account for the `loading="lazy"` attribute if that's enabled.

Do the Automattic tests pass now? I don't see the `matticbot` icon in the builds below.

Also, the [failed Travis tests](https://travis-ci.org/github/Automattic/jetpack/jobs/676436204#L604) don't look to be related to this PR, but I could be wrong.

Reply:
Update: it looks like the `matticbot` says 'Required review' now, but I'm not sure what that means about the tests.

Reply:
> the failed Travis tests don't look to be related to this PR, but I could be wrong

I've rebased the PR to include some changes we had to make earlier today following some changes in Core (see #15494).

Reply:
> Do the Automattic tests pass now?

I'm afraid not.

```
[test_shortcodes_slideshow_amp] test_shortcodes_slideshow_amp with data set "without_autostart_attribute"
[19:26:51][test_shortcodes_slideshow_amp with data set "without_autostart_attribute"] Failed asserting that two strings are equal.
[19:26:51][test_shortcodes_slideshow_amp with data set "without_autostart_attribute"] modules/shortcodes/test-class.slideshow.php:136
 
[19:26:51][test_shortcodes_slideshow_amp with data set "without_autostart_attribute"] '<div class="wp-block-jetpack-slideshow wp-amp-block wp-block-jetpack-slideshow__autoplay wp-block-jetpack-slideshow__autoplay-playing" id="wp-block-jetpack-slideshow__1"><div class="wp-block-jetpack-slideshow_container swiper-container"><amp-carousel width="800" height="600" layout="responsive" type="slides" data-next-button-aria-label="Next Slide" data-prev-button-aria-label="Previous Slide" controls loop autoplay delay=3000 id="wp-block-jetpack-slideshow__amp-carousel__1" on="slideChange:wp-block-jetpack-slideshow__amp-pagination__1.toggle(index=event.index, value=true)"><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/380428471_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/380428471_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/380428471_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/380428471_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/380428471_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/380428471_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/380428471_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/380428471_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/380428471_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/380428471_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/380428471_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/380428471_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/380428471_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/380428471_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div></amp-carousel><a aria-label="Pause Slideshow" class="wp-block-jetpack-slideshow_button-pause" role="button" on="tap:wp-block-jetpack-slideshow__amp-carousel__1.toggleAutoplay(toggleOn=false),wp-block-jetpack-slideshow__1.toggleClass(class=wp-block-jetpack-slideshow__autoplay-playing,force=false)"></a><a aria-label="Play Slideshow" class="wp-block-jetpack-slideshow_button-play" role="button" on="tap:wp-block-jetpack-slideshow__amp-carousel__1.toggleAutoplay(toggleOn=true),wp-block-jetpack-slideshow__1.toggleClass(class=wp-block-jetpack-slideshow__autoplay-playing,force=true)"></a><amp-selector id="wp-block-jetpack-slideshow__amp-pagination__1" class="wp-block-jetpack-slideshow_pagination swiper-pagination swiper-pagination-bullets amp-pagination" on="select:wp-block-jetpack-slideshow__amp-carousel__1.goToSlide(index=event.targetOption)" layout="container"><button option="0" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 1" selected></button><button option="1" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 2" ></button><button option="2" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 3" ></button><button option="3" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 4" ></button><button option="4" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 5" ></button><button option="5" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 6" ></button><button option="6" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 7" ></button><button option="7" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 8" ></button><button option="8" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 9" ></button><button option="9" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 10" ></button><button option="10" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 11" ></button><button option="11" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 12" ></button><button option="12" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 13" ></button><button option="13" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 14" ></button></amp-selector></div></div>'
[19:26:51][test_shortcodes_slideshow_amp with data set "without_autostart_attribute"] '<div class="wp-block-jetpack-slideshow wp-amp-block wp-block-jetpack-slideshow__autoplay wp-block-jetpack-slideshow__autoplay-playing" id="wp-block-jetpack-slideshow__1"><div class="wp-block-jetpack-slideshow_container swiper-container"><amp-carousel width="800" height="600" layout="responsive" type="slides" data-next-button-aria-label="Next Slide" data-prev-button-aria-label="Previous Slide" controls loop autoplay delay=3000 id="wp-block-jetpack-slideshow__amp-carousel__1" on="slideChange:wp-block-jetpack-slideshow__amp-pagination__1.toggle(index=event.index, value=true)"><div class="wp-block-jetpack-slideshow_slide"><figure><img width="800" height="600" src="http://example.org/files/2020/04/380428471_image1.jpg?w=800" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="800" height="600" src="http://example.org/files/2020/04/1643280503_image1.jpg?w=800" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div></amp-carousel><a aria-label="Pause Slideshow" class="wp-block-jetpack-slideshow_button-pause" role="button" on="tap:wp-block-jetpack-slideshow__amp-carousel__1.toggleAutoplay(toggleOn=false),wp-block-jetpack-slideshow__1.toggleClass(class=wp-block-jetpack-slideshow__autoplay-playing,force=false)"></a><a aria-label="Play Slideshow" class="wp-block-jetpack-slideshow_button-play" role="button" on="tap:wp-block-jetpack-slideshow__amp-carousel__1.toggleAutoplay(toggleOn=true),wp-block-jetpack-slideshow__1.toggleClass(class=wp-block-jetpack-slideshow__autoplay-playing,force=true)"></a><amp-selector id="wp-block-jetpack-slideshow__amp-pagination__1" class="wp-block-jetpack-slideshow_pagination swiper-pagination swiper-pagination-bullets amp-pagination" on="select:wp-block-jetpack-slideshow__amp-carousel__1.goToSlide(index=event.targetOption)" layout="container"><button option="0" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 1" selected></button><button option="1" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 2" ></button></amp-selector></div></div>'
[19:26:51][test_shortcodes_slideshow_amp] test_shortcodes_slideshow_amp with data set "with_autostart_attribute"
[19:26:51][test_shortcodes_slideshow_amp with data set "with_autostart_attribute"] Failed asserting that two strings are equal.
[19:26:51][test_shortcodes_slideshow_amp with data set "with_autostart_attribute"] modules/shortcodes/test-class.slideshow.php:136
 
[19:26:51][test_shortcodes_slideshow_amp with data set "with_autostart_attribute"] '<div class="wp-block-jetpack-slideshow wp-amp-block wp-block-jetpack-slideshow__autoplay wp-block-jetpack-slideshow__autoplay-playing" id="wp-block-jetpack-slideshow__2"><div class="wp-block-jetpack-slideshow_container swiper-container"><amp-carousel width="800" height="600" layout="responsive" type="slides" data-next-button-aria-label="Next Slide" data-prev-button-aria-label="Previous Slide" controls loop autoplay delay=3000 id="wp-block-jetpack-slideshow__amp-carousel__2" on="slideChange:wp-block-jetpack-slideshow__amp-pagination__2.toggle(index=event.index, value=true)"><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div></amp-carousel><a aria-label="Pause Slideshow" class="wp-block-jetpack-slideshow_button-pause" role="button" on="tap:wp-block-jetpack-slideshow__amp-carousel__2.toggleAutoplay(toggleOn=false),wp-block-jetpack-slideshow__2.toggleClass(class=wp-block-jetpack-slideshow__autoplay-playing,force=false)"></a><a aria-label="Play Slideshow" class="wp-block-jetpack-slideshow_button-play" role="button" on="tap:wp-block-jetpack-slideshow__amp-carousel__2.toggleAutoplay(toggleOn=true),wp-block-jetpack-slideshow__2.toggleClass(class=wp-block-jetpack-slideshow__autoplay-playing,force=true)"></a><amp-selector id="wp-block-jetpack-slideshow__amp-pagination__2" class="wp-block-jetpack-slideshow_pagination swiper-pagination swiper-pagination-bullets amp-pagination" on="select:wp-block-jetpack-slideshow__amp-carousel__2.goToSlide(index=event.targetOption)" layout="container"><button option="0" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 1" selected></button><button option="1" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 2" ></button><button option="2" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 3" ></button><button option="3" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 4" ></button><button option="4" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 5" ></button><button option="5" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 6" ></button><button option="6" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 7" ></button><button option="7" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 8" ></button><button option="8" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 9" ></button><button option="9" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 10" ></button><button option="10" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 11" ></button><button option="11" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 12" ></button><button option="12" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 13" ></button><button option="13" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 14" ></button><button option="14" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 15" ></button><button option="15" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 16" ></button></amp-selector></div></div>'
[19:26:51][test_shortcodes_slideshow_amp with data set "with_autostart_attribute"] '<div class="wp-block-jetpack-slideshow wp-amp-block wp-block-jetpack-slideshow__autoplay wp-block-jetpack-slideshow__autoplay-playing" id="wp-block-jetpack-slideshow__2"><div class="wp-block-jetpack-slideshow_container swiper-container"><amp-carousel width="800" height="600" layout="responsive" type="slides" data-next-button-aria-label="Next Slide" data-prev-button-aria-label="Previous Slide" controls loop autoplay delay=3000 id="wp-block-jetpack-slideshow__amp-carousel__2" on="slideChange:wp-block-jetpack-slideshow__amp-pagination__2.toggle(index=event.index, value=true)"><div class="wp-block-jetpack-slideshow_slide"><figure><img width="800" height="600" src="http://example.org/files/2020/04/16651216_image1.jpg?w=800" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="800" height="600" src="http://example.org/files/2020/04/1489048507_image1.jpg?w=800" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div></amp-carousel><a aria-label="Pause Slideshow" class="wp-block-jetpack-slideshow_button-pause" role="button" on="tap:wp-block-jetpack-slideshow__amp-carousel__2.toggleAutoplay(toggleOn=false),wp-block-jetpack-slideshow__2.toggleClass(class=wp-block-jetpack-slideshow__autoplay-playing,force=false)"></a><a aria-label="Play Slideshow" class="wp-block-jetpack-slideshow_button-play" role="button" on="tap:wp-block-jetpack-slideshow__amp-carousel__2.toggleAutoplay(toggleOn=true),wp-block-jetpack-slideshow__2.toggleClass(class=wp-block-jetpack-slideshow__autoplay-playing,force=true)"></a><amp-selector id="wp-block-jetpack-slideshow__amp-pagination__2" class="wp-block-jetpack-slideshow_pagination swiper-pagination swiper-pagination-bullets amp-pagination" on="select:wp-block-jetpack-slideshow__amp-carousel__2.goToSlide(index=event.targetOption)" layout="container"><button option="0" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 1" selected></button><button option="1" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 2" ></button></amp-selector></div></div>'

```